python/test_python.py

```diff
@@ -199,4 +200,25 @@ class TestEncodeBase64:
     def test_encode(self, docker_runner):
         docker_runner.run('python encode.py "Hello, world!"')
         docker_runner.container.wait()
-        assert str(docker_runner.container.logs(), 'UTF-8') == 'SGVsbG8sIHdvcmxkIQ==\n'
\ No newline at end of file
+        assert str(docker_runner.container.logs(), 'UTF-8') == 'SGVsbG8sIHdvcmxkIQ==\n'
+
+class TestStreamingStdin:
+    """Test that streaming stdin can be read line by line and can write to stdout
+    without waiting for all lines to arrive
+    Note: this test uses Docker CLI instead of the Python Docker SDK (implemented in the
+    `docker_runner` fixture) since SDK doesn't easily allow for streaming inputs"""
+    def test_stdin(self):
+        # Start a subprocess that runs the script and waits for input
+        script = subprocess.Popen(
+            ['docker', 'run', '-i', 'python-rosetta', 'python', 'streaming_stdin.py'],
```

Comment:
This test assumes that the python-rosetta image exists whereas the other tests use the fixture which builds the image fresh. Is this ok?

Reply:
Yeah, we should use the `docker_image` fixture to make sure the image gets built. In fact, the fixture returns the image name as a string, so we could use that instead of hard-coding the image name in this test.

Reply:
Oh right, that makes sense! Yay for versatile fixtures!